battleship-logic/src/main/java/ship/Ship.java

```diff
@@ -40,6 +40,7 @@ public boolean isDamaged() {
         return (size - damaged) == 0;
     }
 
+    // FIXME the equal and hash must be carefully implemented. Cus more ships could have the same name size and damaged.
```

Comment:
Ok, now I know what is a point. What is better, use UUID.randomId, or create own variable of id?